### PR TITLE
Add BaseEntity Class

### PR DESCRIPTION
**Pull request by zysim**
* Adds `created_at`, `updated_at`, and `deleted_at` fields to all models
* Updates `Equals` methods to match all above timestamp fields
* Updates DeleteModship to soft-delete, along with its test

### CODE REVIEW

File: LeaderboardBackend/Models/Entities/BaseEntity.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+using System.ComponentModel.DataAnnotations;
+using NodaTime;
+
+/// <summary>
+///     Base class that provides common fields to all Entities.
+/// </summary>
+public class BaseEntity
```
Comment (ayecptn) [inline LeaderboardBackend/Models/Entities/BaseEntity.cs:7]: As mentioned in #129 make this class abstract

File: LeaderboardBackend/Models/Entities/BaseEntity.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	/// </summary>
+	[Required]
+	public Instant UpdatedAt { get; set; } = SystemClock.Instance.GetCurrentInstant();
+
+	/// <summary>
+	///     The time this entity was deleted.
+	/// </summary>
+	public Instant? DeletedAt { get; set; }
+
+	public override bool Equals(object? obj)
```
Comment (ayecptn) [inline LeaderboardBackend/Models/Entities/BaseEntity.cs:26]: Make the Equals and GetHashCode also abstract and implemented them in each Entity

**Review by ayecptn — CHANGES_REQUESTED** [review]

File: LeaderboardBackend/Models/Entities/BaseEntity.cs (hunk 2 above)
Comment (zysim) [reply LeaderboardBackend/Models/Entities/BaseEntity.cs:26]: Actually, I just need to remove these lines in `BaseEntity`, as they already are overrides here. The methods are part of the inner `object` type. I'll still do the latter part, of course.

**Comment by zysim** [comment]
Stale PR. Closing.

**Pull request closed without merging**